### PR TITLE
manifest: mbed TLS 3.0.0 afterwork

### PR DESCRIPTION
**Pull request by frkv**
-Updating to nrfxlib pr [#637](https://github.com/nrfconnect/sdk-nrfxlib/pull/637)
-Updating to Zephyr pr [#709](https://github.com/nrfconnect/sdk-zephyr/pull/709)
-Updating to sdk-mcuboot pr [#195](https://github.com/nrfconnect/sdk-mcuboot/pull/195)
-Updating to sdk-trusted-firmware-m pr [#38](https://github.com/nrfconnect/sdk-trusted-firmware-m/pull/38)

-Improves code-usage and mbed TLS 3.0.0 support
-Adds cc3xx with PSA support

ref: NCSDK-13563
ref: NCSDK-13503
ref: NCSDK-13530
ref NCSDK-13212

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

test-sdk-nrf: sdk-nrf-mbedtls3-afterwork

### CODE REVIEW

File: applications/asset_tracker_v2/doc/app_module.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+.. _asset_tracker_v2_app_module:
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/doc/app_module.rst:1]: There are unrelated changes in the PR.

Comment (frkv) [reply applications/asset_tracker_v2/doc/app_module.rst]: This was because we were pulling in a PR that wasn't in yet

File: modules/trusted-firmware-m/Kconfig.psa.defconfig [hunk 2]
```diff
@@ -97,7 +97,7 @@ config PSA_WANT_ALG_OFB
 	default y
 config PSA_WANT_ALG_XTS
 	bool
-	default y
+	#default y -- Currently not supported
```
Comment (tejlmand) [inline modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: this doesn't remove the setting, only changing its default.

Other part can still do a `select PSA_WANT_ALG_XTS`.
I do notice it's not use configurable, as it is promptless.

Secondly, we should not have code commented out like this, better to remove, a new commit can always re-introduce the default later.

**Review by tejlmand — COMMENTED** [review]
minor observation

File: modules/trusted-firmware-m/Kconfig.psa.defconfig (hunk 2 above)
Comment (frkv) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: This is a defconfig, the real configuration also has a removed prompt. The problem was that the defconfig made this a valid configuration (when it currently isn't supported)

We will rectify this missing feature in upcoming work (after nRF Connect SDK 1.9.0 release)

Comment (tejlmand) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: so then reintroduce the setting in the defconfig at that time.

I would prefer not to have a setting doing nothing in this file.

Comment (frkv) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: This file contains a one to one match of all  available configurations in the order that they are presented in mbed TLS PSA configuration. We are going for completeness to not hide away missing features, so I disagree on this

Comment (SebastianBoe) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: What about a comment that says

# PSA_WANT_ALG_XTS is currently not supported.


In any case, we can't have commented code like this

Comment (frkv) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: Accepted. Will rework the whole file for all items like this!

File: modules/tfm/tfm/boards/CMakeLists.txt [hunk 3]
```diff
@@ -51,7 +51,7 @@ endif()
 
 if (${TFM_PARTITION_CRYPTO})
   target_link_libraries(platform_s PRIVATE platform_cc3xx)
-  target_link_libraries(platform_s PRIVATE crypto_service_mbedtls)
+  target_link_libraries(platform_s PRIVATE crypto_service_mbedcrypto)
```
Comment (joerchan) [inline modules/tfm/tfm/boards/CMakeLists.txt:54]: This was a workaround. You can remove the line.

File: modules/trusted-firmware-m/Kconfig.psa.defconfig (hunk 2 above)
Comment (joerchan) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]:
```suggestion
	# PSA_WANT_ALG_XTS is currently not supported.
```

**Review by joerchan — APPROVED** [review]

File: modules/tfm/tfm/boards/CMakeLists.txt (hunk 3 above)
Comment (SebastianBoe) [reply modules/tfm/tfm/boards/CMakeLists.txt:54]: Removed the line. But I'm awaiting applying the patch so we don't cancel the CI run.

File: modules/trusted-firmware-m/Kconfig.psa.defconfig (hunk 2 above)
Comment (SebastianBoe) [reply modules/trusted-firmware-m/Kconfig.psa.defconfig:100]: Removed the config and replaced it with a comment. But I'm not pushing the fix because we don't want to cancel the CI run.

**Review by SebastianBoe — APPROVED** [review]
all feedback has been addressed

**Review by tejlmand — APPROVED** [review]
lgtm

**Review by shantha-14 — APPROVED** [review]
